test/NuGet.Clients.Tests/NuGet.Credentials.Test/SecurePluginCredentialProviderTests.cs

```diff
@@ -0,0 +1,198 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using Newtonsoft.Json.Linq;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.Plugins;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.Credentials.Test
+{
+    public class SecurePluginCredentialProviderTests
+    {
+        [Fact]
+        public void Create_ThrowsForNullPlugin()
+        {
+            var exception = Assert.Throws<ArgumentNullException>(
+                () => new SecurePluginCredentialProvider(null, NullLogger.Instance));
+
+            Assert.Equal("pluginDiscoveryResult", exception.ParamName);
+        }
+
+        [Fact]
+        public void Create_ThrowsForNullLogger()
+        {
+            var exception = Assert.Throws<ArgumentNullException>(
+                () => new SecurePluginCredentialProvider(CreatePluginDiscoveryResult(), null));
+
+            Assert.Equal("logger", exception.ParamName);
+        }
+
+        [Fact]
+        public void Create_ThrowsForInvalidPlugin()
+        {
+            var exception = Assert.Throws<ArgumentException>(
+                () => new SecurePluginCredentialProvider(CreatePluginDiscoveryResult(PluginFileState.InvalidFilePath), NullLogger.Instance));
+        }
+
+        [Fact]
+        public void Type_IsICredentialProvider()
+        {
+            var provider = new SecurePluginCredentialProvider(CreatePluginDiscoveryResult(), NullLogger.Instance);
+            Assert.True(provider is ICredentialProvider);
+        }
+
+        [Fact]
+        public void Provider_IdContainsPath()
+        {
+            var pluginResult = CreatePluginDiscoveryResult();
+            var provider = new SecurePluginCredentialProvider(pluginResult, NullLogger.Instance);
+            Assert.Contains(pluginResult.PluginFile.Path, provider.Id);
+        }
+
+
+        private static PluginDiscoveryResult CreatePluginDiscoveryResult(PluginFileState pluginState = PluginFileState.Valid)
+        {
+            return new PluginDiscoveryResult(new PluginFile(@"C:\random\path\plugin.exe", pluginState));
+        }
+
+        [PlatformFact(Platform.Windows)]
+        public void TryCreate_ReturnsValidCredentials()
+        {
+            var uri = new Uri("https://api.nuget.org/v3/index.json");
+            var authUsername = "username";
+            var authPassword = "password";
+            var expectation = new TestExpectation(
+                null,
+                null,
+                new[] { OperationClaim.Authentication },
+                ConnectionOptions.CreateDefault(),
+                Protocol.Plugins.ProtocolConstants.CurrentVersion,
+                uri,
+                authUsername,
+                authPassword,
+                true,
+                true
```

Comment:
Add named parameters to all these pesky constant literal arguments.  `null` and `true` aren't very readable.  This feedback applies to subsequent tests as well.